portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,41 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+//    public ArrayList<String> comments = new ArrayList<String>(List.of("boom", "bam", "pow"));
+
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // obtain input from form(the user's comment)
+    String userComment = getParameter(request, "userCommentInput", "");
+    
+    // seperate words into indivdual text
+    String[] words = userComment.split("\\s*,\\s*");
+
+    // Respond with the result.
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println(Arrays.toString(words));
+  }
+  
+//   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+//     // Get the input from the form.
+//     response.setContentType("text/html;");
+//     String userComment = getParameter(request, "userCommentInput", "");
+//     response.getWriter().println(userComment);
+//   }
+  
+  public String convertToJsonUsingGson(ArrayList<String> list) {
+    Gson gson = new Gson();
+    String userComment = gson.toJson(list);
+    return userComment;
+  }
+    
+    //requst parameter was not specified by client
+  public String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
```

Comment:
both of these methods are not used?

Please try to make sure a pull request has a good purpose and contains code only for that purpose. Keep un related code out not only will help the code reviewer, but also make management easier (e.g. rollbacks, merges etc).